### PR TITLE
fcast-receiver: init at 1.0.14; nixos/fcast-receiver: init module

### PR DESCRIPTION
**Pull request by ymstnt**
## Description of changes

- [FCast Receiver](https://gitlab.futo.org/videostreaming/fcast/-/tree/master/receivers/electron) is a receiver application for an open-source media streaming protocol, [FCast](https://fcast.org/), which is an alternative to Chromecast and AirPlay.
- The program requires port `46899` open in the firewall to function correctly, so I've created a module that has an `openFirewall` option that opens the port for the user.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/fc/fcast-receiver/package.nix [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  env.ELECTRON_SKIP_BINARY_DOWNLOAD = "1";
+
+  desktopItems = [
+    (makeDesktopItem {
+      name = pname;
+      desktopName = "FCast Receiver";
+      genericName = "Media Streaming Receiver";
+      exec = "fcast-receiver";
+      icon = "fcast-receiver";
+      comment = meta.description;
```
Comment (SuperSandro2000) [inline pkgs/by-name/fc/fcast-receiver/package.nix:38]: Changes to meta should not trigger rebuilds

File: pkgs/by-name/fc/fcast-receiver/package.nix [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    owner = "videostreaming";
+    repo = "fcast";
+    rev = "b13d0f7e8150c279d377a78f89d338b7fc0f5539";
+    hash = "sha256-NqOQFEP5MWFNWHj6ZpNCVh8D/4YlNhqA/Db6hdAXepo=";
+  };
+
+  sourceRoot = "${src.name}/receivers/electron";
+
+  makeCacheWritable = true;
+  dontPatchELF = true;
```
Comment (SuperSandro2000) [inline pkgs/by-name/fc/fcast-receiver/package.nix:25]: Why not?

Comment (YMSTNT) [reply pkgs/by-name/fc/fcast-receiver/package.nix]: I've cheked, and removed dontPatchELF as it was not needed, probably remained in from testing. makeCacheWritable was needed however, package failed to build without that

**Comment by YMSTNT** [comment]
I've corrected the issues from the review. 
Also bumped version to 1.0.14, several fixes have been made to the QR code pairing since 1.0.12.

**Pull request merged**